W-16139934 AEH sso restructuring and azure content jh (#49)
mulesoft/docs-api-experience-hub: modules/ROOT/pages/gathering-set-up-information.adoc

= Gathering Setup Information for SSO
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Use the following sections to gather the information you'll need to set up SSO.

Reviewer: The short description should summarize, not introduce, the topic. Can you provide a quick summary of the information needed?

> Use the following sections to gather the information you'll need to set up SSO.

Reviewer: ```suggestion
Use these sections to gather the information needed to set up SSO.
```

You might need to collaborate with your security team or other team members if you don't have access to this information. For more information, see xref:user-roles-permissions-and-profiles.adoc#user-roles-for-configuring-sso[User Roles for Configuring SSO].

[[username-strategy]]
== Username Assignment Strategy for a Shared Identity

> When an identity provider is already configured for your Anypoint Platform organization, review the username assignment strategy since it can be different for the portal and Anypoint Platform users. 

Reviewer: ```suggestion
If an identity provider is configured for your Anypoint Platform organization, review the username assignment strategy because it can be different for the portal and Anypoint Platform users. 
```

> To have a single identity between the portal and Anypoint Platform, the format for the username assignment must be the same. If the username strategy does not match, you must reconcile the username format by adding attributes for *anypoint_idp_id* and *anypoint_username*. Then, this information must be mapped with the Anypoint Platform identity provider that you are using for SSO. 

Reviewer: ```suggestion
To have a single identity between the portal and Anypoint Platform, the format for the username assignment must be the same. If the username strategy doesn’t match, reconcile the username format by adding attributes for *anypoint_idp_id* and *anypoint_username*. Then, map this information to the Anypoint Platform identity provider that you are using for SSO. 
```

Reviewer: Bold correct?

> To have a single identity between the portal and Anypoint Platform, the format for the username assignment must be the same. If the username strategy does not match, you must reconcile the username format by adding attributes for *anypoint_idp_id* and *anypoint_username*. Then, this information must be mapped with the Anypoint Platform identity provider that you are using for SSO. You add these custom attributes using the identity provider's profile editor. 

Reviewer 2: suggestion: `profile editor` is a something specific for okta.
Maybe we should say something more generic here or remove the last  sentence.

Author: I removed that sentence

> For more information about adding custom claims when the identity provider is already configured but not in the required format, see the *Required Claims and Attributes for SSO* section. 

Reviewer: Re: the *Required Claims and Attributes for SSO* section

Is this a section in the UI or in the doc? If doc, suggest "see _Required Claims and Attributes for SSO_. Or make it a link.

== Required Claims and Attributes for SSO 

> Be aware of what API Experience Hub requires for claims or attributes when creating a new application for the portal in the identity provider.

Reviewer: ```suggestion
Be aware of API Experience Hub requirements for claims or attributes when creating a new application for the portal in the identity provider.
```

Configuring the required user claims in the identity provider ensures that the user information is mapped correctly to Salesforce and Anypoint Platform users.  

> The AEHPortalRegistrationHandler requires specific user information from the identity provider to create or update a user. The registration handler uses required claims to extract user information. The following table lists the required information and which claims the information is extracted from. If there is more than one claim for the same value, the registration handler looks for the first expected claim in the order it appears in the table:

Reviewer: Font for AEHPortalRegistrationHandler?

Reviewer: Global: search for "the following" and replace with "this" or "these".

Reviewer: Avoid "there is" - suggest "If more than one claim exists"

[%header%autowidth.spread]
|===
|Field|Expected Claims
|First Name 
| given_name, first_name, firstname, and firstName
|Last Name 
| family_name, last_name, lastname, and lastName
|Email 
|email, email_address, and emailAddress
|Username 
|preferred_username, federation_identifier, email, sub, and NameId
|Groups | groups
|===
 
To configure SSO when the identity provider is already configured for Anypoint Platform, use this information:

[%header,cols="1a,2a,3a"]
|===
|Field |Description |Expected Values
|Anypoint Idp Id|Map identities to the specific identity provider in Anypoint Platform. |anypoint_idp_id

> |Anypoint Username|Explicitly specify the Anypoint Platform user when the user's identity already exists in Anypoint Platform, but the username is different than the Salesforce user. |anypoint_username

Reviewer: ```suggestion
|Anypoint Username|Explicitly specify the Anypoint Platform user if the user's identity exists in Anypoint Platform, but the username is different than the Salesforce user. |anypoint_username
```
